src/controllers/ioWalletController.ts

```diff
@@ -33,3 +50,97 @@ export const retrieveUserId = (
       )
     )
   );
+
+export default class IoWalletController {
+  constructor(private readonly ioWalletService: IoWalletService) {}
+
+  /**
+   * Get nonce
+   */
+  public readonly getNonce = (): Promise<
+    IResponseErrorInternal | IResponseSuccessJson<NonceDetailView>
+  > => this.ioWalletService.getNonce();
+
+  /**
+   * Create a Wallet Instance
+   */
+  public readonly createWalletInstance = (
+    req: express.Request
+  ): Promise<
+    | IResponseErrorInternal
+    | IResponseErrorGeneric
+    | IResponseErrorValidation
+    | IResponseSuccessJson<undefined>
+  > =>
+    withUserFromRequest(req, async (user) =>
+      pipe(
+        sequenceS(TE.ApplyPar)({
+          body: pipe(
+            req.body,
+            CreateWalletInstanceBody.decode,
+            E.mapLeft(() =>
+              ResponseErrorInternal("Error validating the request body")
```

Comment:
here you're swallowing validation errors, maybe something like 

```suggestion
              ResponseErrorInternal("Error validating the request body: ${readableReport(errors)}")
```